code/modules/mob/living/living.dm

```diff
@@ -311,6 +311,10 @@
 		if (C.legcuffed && !initial(C.legcuffed))
 			C.unEquip(C.legcuffed)
 		C.legcuffed = initial(C.legcuffed)
+		if(C.reagents)
+			for(var/datum/reagent/R in C.reagents.reagent_list)
+				C.reagents.clear_reagents()
+			C.reagents.addiction_list = list()
```

Comment:
I was wondering that myself as I know this can, under some instances, cause leaks--will change it.